services/dedup/badger/badger_test.go

```diff
@@ -0,0 +1,51 @@
+package badger
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/services/dedup/types"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+func Test_Badger(t *testing.T) {
+	config.Reset()
+	logger.Reset()
+	misc.Init()
+
+	dbPath := os.TempDir() + "/dedup_test"
+	defer func() { _ = os.RemoveAll(dbPath) }()
+	_ = os.RemoveAll(dbPath)
+	conf := config.New()
+	t.Setenv("RUDDER_TMPDIR", dbPath)
+	badger := NewBadgerDB(conf, stats.NOP, DefaultPath())
+	require.NotNil(t, badger)
+	defer badger.Close()
+	t.Run("Same messageID should be deduped from badger", func(t *testing.T) {
+		key1 := types.KeyValue{Key: "a", Value: 1, WorkspaceId: "test"}
+		key2 := types.KeyValue{Key: "a", Value: 1, WorkspaceId: "test"}
+		found, _, err := badger.Get(key1)
```

Comment:
```suggestion
		notAvailable, _, err := badger.Get(key1)
```
its not found, we are returning false if key is seen earlier